Development/cpprest/http_utils.cpp

```diff
@@ -330,6 +330,111 @@ namespace web
                 return result;
             }
 
+            utility::string_t make_directives_header(const directives& values)
+            {
+                utility::string_t result;
+                for(auto it = values.begin(); it != values.end(); ++it)
+                {
+                    if (it != values.begin()) { result.push_back(U(';')); }
+                    result.append(it->first);
+                    if (!it->second.empty())
+                    {
+                        result.push_back(U('='));
+                        result.append(it->second);
+                    }
+                }
+                return result;
+            }
+
+            directives parse_directives_header(const utility::string_t& value)
+            {
+                enum {
+                    pre_param,
+                    pre_param_name,
+                    param_name,
+                    pre_param_value,
+                    param_value,
+                    param_value_token,
+                    param_value_quoted_string,
+                    param_value_quoted_string_escape
+                } state = pre_param_name;
+
+                // token      = 1*<any CHAR except CTLs or separators>
+                // separators = "(" | ")" | "<" | ">" | "@"
+                //            | "," | ";" | ":" | "\" | <">
+                //            | "/" | "[" | "]" | "?" | "="
+                //            | "{" | "}" | SP | HT
+                // see https://datatracker.ietf.org/doc/html/rfc2616#section-2.2
```

Comment:
Since RFC 2616 is obsoleted by RFC 7230, I thought it was worth checking the definition of `token` and `tchar` there, and for which we already had the `is_tchar` function above.
https://tools.ietf.org/html/rfc7230#section-3.2.6
I believe the two specifications are equivalent, so despite RFC 6797 referring to the older RFC 2616, would it be better to reuse the existing function?